Backend/app/kafka-consumers/src/Consumer/CustomerStats/Processor.hs

```diff
@@ -0,0 +1,74 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Consumer.CustomerStats.Processor
+  ( updateCustomerStats,
+  )
+where
+
+import "rider-app" Domain.Types.BookingCancellationReason as SBCR
+import "dynamic-offer-driver-app" Domain.Types.Ride as DDR
+import Environment
+import Kernel.Beam.Functions
+import Kernel.Prelude
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import "sessionizer-metrics" Lib.SessionizerMetrics.Types.Event as E
+import "rider-app" SharedLogic.Person as SP
+import "rider-app" Storage.CachedQueries.Merchant as CQMerchant
+import "rider-app" Storage.Queries.BookingCancellationReason as QBCR
+import "rider-app" Storage.Queries.Person.PersonStats as QP
+import "rider-app" Tools.Error
+import "dynamic-offer-driver-app" Tools.Event as TE
+
+updateCustomerStats :: E.Event TE.Payload -> Text -> Flow ()
+updateCustomerStats event _ = do
+  whenJust event.personId $ \pId -> do
+    when (event.service == RIDER_APP) do
+      let personId = Id pId
+          merchantId = Id event.merchantId
+      -- personStats <- Esq.runInReplica $ QP.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+      personStats <- runInReplica $ QP.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+      when (isNotBackfilled personStats) do
+        personStatsValues <- SP.backfillPersonStats personId merchantId
```

Comment:
Person having zero rides and keeps calling this API will result in a lot of useless queries in `backfillPersonStats`. We can check from ride table if the driver has taken any ride and then only choose to call this.

Reply:
but the event will be in kafka only when person is going to take a ride, right ?

Reply:
Yes but I'm talking about the `personStats` API, same logic is used there also